src/main/java/com/gojek/beast/consumer/MessageConsumer.java

```diff
@@ -44,22 +44,24 @@ public Status consume() throws WakeupException {
             return new SuccessStatus();
         }
         log.info("Pulled {} messages", messages.count());
-        Status status = pushToSink(messages);
+        Status status = pushToSink(messages, startTime);
         statsClient.timeIt("kafkaConsumer.consumption.time", startTime);
         return status;
     }
 
-    private Status pushToSink(ConsumerRecords<byte[], byte[]> messages) {
+    private Status pushToSink(ConsumerRecords<byte[], byte[]> messages, Instant pollTime) {
         List<Record> records;
         try {
+            final Instant deSerTime = Instant.now();
```

Comment:
Use clock class instead of Instant.now()

Reply:
As discussed: existing code base uses now() extensively. Will be huge refactoring across board.